source/server/server.cc

```diff
@@ -55,7 +55,7 @@ InstanceImpl::InstanceImpl(Options& options, Network::Address::InstanceConstShar
     try {
       Logger::Registry::getSink()->logToFile(options.logPath(), access_log_manager_);
     } catch (const EnvoyException& e) {
-      PANIC(
+      throw EnvoyException(
```

Comment:
What do you think about just moving this whole block into the try/catch below? Then we only have 1 place where we catch init issues (I don't want to catch exceptions thrown from run()). Then you can just do a proper death test?

Reply:
I had it at the top to switch to the desired logging location as early as possible, but I didn't look to see what all actually logs in this function.

There's also another throw a few lines down (checking git SHA), so there are a few exceptions that can escape up to main.  What about if I move the try that starts before initialize to the top of this constructor, so the whole thing is in the try/catch?

Or I can just move this logging change into the existing try/catch without expanding the scope.  I'm not too worried about a handful of log messages that might occur in those few lines.

Reply:
It's up to you. I mostly don't want to include run() in the try/catch block. I think it's fine to move the try/catch to start to the top of the constructor. That makes sense to me. Then you can evert the main changes and do a proper death test.